linkcheck/checker/urlbase.py

```diff
@@ -663,7 +664,8 @@ def get_user_password (self):
         """
         if self.userinfo:
             # URL itself has authentication info
-            return urllib.parse.splitpasswd(self.userinfo)
+            o = urllib.parse.urlparse(self.userinfo)
+            return (o.username, o.password)
```

Comment:
This fells wrong.

```
>>> urllib.parse.urlparse('user:passwd')
ParseResult(scheme='user', netloc='', path='passwd', params='', query='', fragment='')
>>> urllib.parse.urlparse('user:passwd').username
>>> urllib.parse.urlparse('user:passwd').password
```
```suggestion
            o = urllib.parse.urlparse('//{}@'.format(self.userinfo))
            return (o.username, o.password)
```

Reply:
Glad you caught that. I've added the change.

Reply:
What worries me is that no unit test caught this.

Let's add one?

Reply:
I should probably insist, but I'm so tired I can't bring myself to care.

Reply:
don't worry about it then. :)